pkg/util/httprequest/httprequest.go

```diff
@@ -68,14 +66,6 @@ func SchemeHost(req *http.Request) (string /*scheme*/, string /*host*/) {
 
 	forwardedProto := forwarded("Proto")
 	forwardedHost := forwarded("Host")
-	// If both X-Forwarded-Host and X-Forwarded-Port are sent, use the explicit port info
-	if forwardedPort := forwarded("Port"); len(forwardedHost) > 0 && len(forwardedPort) > 0 {
```

Comment:
> But if there is no port in the Host header and X-Forwarded-Port is a default port, then the port is not added to the Host header.

It should only be exluded if it is a default port *that matches the scheme* (80 && http, 443 && https)